vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,201 @@
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+			*/
+			"advanced_networking": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "True if the gateway uses advanced networking",
+			},
+			"backing_config": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Configuration of the vShield edge VM for this gateway. One of: compact, full",
+			},
+			"ha_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "Enable high availability on this edge gateway",
+			},
+			"external_networks": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of external networks to be used by the edge gateway",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"default_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
+				Description: "External network to be used as default gateway. Its name must be included in 'external_networks'. An empty value will skip the default gateway",
+			},
+			"distributed_routing": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "If advanced networking enabled, also enable distributed routing",
+			},
+		},
+	}
+}
+
+// Creates a new edge gateway from a resource definition
+func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] edge gateway creation initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	rawExternalNetworks := d.Get("external_networks").([]interface{})
+	var externalNetworks []string
+	for _, en := range rawExternalNetworks {
+		externalNetworks = append(externalNetworks, en.(string))
+	}
+
+	var gwCreation = govcd.EdgeGatewayCreation{
+		ExternalNetworks:          externalNetworks,
+		Name:                      d.Get("name").(string),
+		OrgName:                   d.Get("org").(string),
+		VdcName:                   d.Get("vdc").(string),
+		Description:               d.Get("description").(string),
+		BackingConfiguration:      d.Get("backing_config").(string),
+		AdvancedNetworkingEnabled: d.Get("advanced_networking").(bool),
+		DefaultGateway:            d.Get("default_gateway").(string),
+		DistributedRoutingEnabled: d.Get("distributed_routing").(bool),
+		HAEnabled:                 d.Get("ha_enabled").(bool),
+	}
+
+	edge, err := govcd.CreateEdgeGateway(vcdClient.VCDClient, gwCreation)
+	if err != nil {
+		log.Printf("[DEBUG] Error creating edge gateway: %#v", err)
+		return fmt.Errorf("error creating edge gateway: %#v", err)
+	}
+
+	d.SetId(d.Get("name").(string))
```

Comment:
Here it is name, while read stores `id=https://192.168.1.109/api/admin/edgeGateway/f2c007f5-ac52-4f57-8a57-b6a1f261a875`
I'd vote for going to real edge gateway ID (not name and not full url)

Reply:
Done. Please see [my note below](https://github.com/terraform-providers/terraform-provider-vcd/pull/263#discussion_r293490824) for some caveats.